snuba/request/schema.py

```diff
@@ -94,18 +84,24 @@ def build_with_extensions(
     ) -> RequestSchema:
         if language == Language.SNQL:
             generic_schema = SNQL_QUERY_SCHEMA
+            extensions_schemas = {}
         else:
             generic_schema = GENERIC_QUERY_SCHEMA
-
-        extensions_schemas = {
-            extension_key: extension.get_schema()
-            for extension_key, extension in extensions.items()
-        }
+            extensions_schemas = {
+                extension_key: extension.get_schema()
+                for extension_key, extension in extensions.items()
+            }
 
         settings_schema = SETTINGS_SCHEMAS[settings_class]
         return cls(generic_schema, settings_schema, extensions_schemas, settings_class)
 
-    def validate(self, value, dataset: Dataset, referrer: str) -> Request:
+    def validate(
+        self,
+        value: MutableMapping[str, Any],
+        dataset: Dataset,
+        referrer: str,
+        language: Language,
```

Comment:
If you build a RequestSchema by passing the language in the factory method (build_with_schema) you do not need to pass it again in the the validate method. This opens the door to misuse. Could you please store the language in an attribute of RequestSchema and reuse it from the validate method?